src/services/identifierService.ts

```diff
@@ -0,0 +1,231 @@
+import { AxiosInstance } from 'axios';
+import {
+  IdentifierCreateReq,
+  Identifier as IdentifierRsp,
+  GenericRsp,
+  IdentifierType,
+  IdentifierUpdateReq,
+  IdentifierStatus,
+  IdentifiersApi,
+  IdentifierList,
+} from '../generated/index.js';
+import { Assert, Helper, isErrorRsp } from '../helpers';
+import httpStatusCodes from '../errors/httpStatusCodes.js';
+import BaseError from '../errors/baseError.js';
+
+export interface IdentifierInterface {
+  create(userId: string, req: IdentifierCreateReq): Promise<IdentifierRsp>;
```

Comment:
I would try to keep the number of exposed functions small (every function we expose could confuse a user.)

I would remove these two functions because they are just nice to have:
- existsByValueAndType
- listAllEmailsByUserId
- listByValueAndType

Every "list" function should support paging optional parameters:

I would remove the WithPaging suffix for all "list" calls (if we add paging as an optional feature to every list function we dont have to include it into each function name).

I would remove the "all" from the function name (it's unclear what it means => list already implies that more that one result can be returned)

So I think we need these list calls:
- list(filter?: string[], sort?: string, page?: number, pageSize?: number): Promise<IdentifierList>;
- listByValueAndType(value: string, type: IdentifierType, sort?: string, page?: number, pageSize?: number): Promise<IdentifierList>;
- listByUserId(userId: string, sort?: string, page?: number, pageSize?: number): Promise<IdentifierList>;
- listByUserIdAndType(userId: string, type: IdentifierType, sort?: string, page?: number, pageSize?: number): Promise<IdentifierList>;